libs/interfaces/src/lib/chains.ts

```diff
@@ -134,6 +135,12 @@ export interface TxInterface<
     tx: Tx | SignedTx,
     opts?: SendOpts
   ) => Promise<Pair<SendResult, ExplorerLink>>;
+  buildParamsFromGenericTx: (
+    params: GenericTxParams,
+    recipientPk: PublicKey<secp256k1>,
+    senderPkSecp: PublicKey<secp256k1>,
+    senderPkEd: PublicKey<ed25519>
+  ) => Promise<BuildTxParams>;
 }
 
 // utility for going to/from key BAF Wallet unified types
```

Comment:
*froths at the mouth* - can we add a converter method for going from `GenericTx` to the chain-specific `Tx` type? Or does that not make sense?

Reply:
Idk, if it makes sense cause you can j chain 2 calls together. I think this keeps the code a bit leaner and also will allow for a very cool modularity thing up and coming in a PR for sending near specific tx via approve redirect